website/docs/r/managed_application.html.markdown

```diff
@@ -0,0 +1,119 @@
+---
+subcategory: "Managed Applications"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_managed_application"
+description: |-
+  Manages a Managed Application.
+---
+
+# azurerm_managed_application
+
+Manages a Managed Application.
+
+## Example Usage
+
+```hcl
+provider "azurerm" {
+  features {}
+}
+
+data "azurerm_client_config" "current" {}
+
+data "azurerm_role_definition" "builtin" {
+  name = "Contributor"
+}
+
+resource "azurerm_resource_group" "example" {
+  name     = "example-resources"
+  location = "West Europe"
+}
+
+resource "azurerm_managed_application_definition" "example" {
+  name                = "example-managedapplicationdefinition"
+  location            = azurerm_resource_group.example.location
+  resource_group_name = azurerm_resource_group.example.name
+  lock_level          = "ReadOnly"
+  package_file_uri    = "https://github.com/Azure/azure-managedapp-samples/raw/master/Managed Application Sample Packages/201-managed-storage-account/managedstorage.zip"
+  display_name        = "TestManagedAppDefinition"
+  description         = "Test Managed App Definition"
+
+  authorization {
+    service_principal_id = data.azurerm_client_config.current.object_id
+    role_definition_id   = split("/", data.azurerm_role_definition.builtin.id)[length(split("/", data.azurerm_role_definition.builtin.id)) - 1]
+  }
+}
+
+resource "azurerm_managed_application" "example" {
+  name                       = "example-managedapplication"
+  location                   = azurerm_resource_group.example.location
+  resource_group_name        = azurerm_resource_group.example.name
+  kind                       = "ServiceCatalog"
+  target_resource_group_name = "infrastructureGroup"
+  application_definition_id  = azurerm_managed_application_definition.example.id
+
+  parameters = {
+    location                 = azurerm_resource_group.example.location
+    storageAccountNamePrefix = "storeNamePrefix"
+    storageAccountType       = "Standard_LRS"
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) Specifies the name of the Managed Application. Changing this forces a new resource to be created.
+
+* `resource_group_name` - (Required) The name of the Resource Group where the Managed Application should exist. Changing this forces a new resource to be created.
+
+* `location` - (Required) Specifies the supported Azure location where the resource exists. Changing this forces a new resource to be created.
+
+* `kind` - (Required) The kind of the managed application. Possible values are `MarketPlace` and `ServiceCatalog`.
+
+* `target_resource_group_name` - (Required) The name of the target resource group which includes resources managed by Managed Application.
```

Comment:
Did you consider naming this `managed_resource_group_name`? I haven't heard this to be referred to as 'target resource group' yet and "managed resource group" would be consistent with `az` as well:

```
$ az managedapp create --help

Command
    az managedapp create : Create a managed application.

Arguments
    --kind              [Required] : The managed application kind. can be marketplace or
                                     servicecatalog.
    --managed-rg-id -m  [Required] : The resource group managed by the managed application.
    --name -n           [Required] : The managed application name.
    --resource-group -g [Required] : The resource group of the managed application.
...
```

Reply:
@katbyte . I think what seiffert suggests is reasonable.  I found generally it would be named as "managed_resource_group_name" in Azure side after went through the related docs. I think it would be more friendly to azure users. How do you think? If it's fine, I will update it to "managed_resource_group_name"?

Reply:
I don't think the API's name is good or clear as to what it is (or if it should be created first) but i'll leave it up to you @neil-yechenwei - please make sure the docs are clear what it means either way

the CLi tool is auto generated from swagger and often carries over tragic names & descriptions @seiffert - we often will change/rename properties so they is more obvious as to what they do ie target resource group implies the resource will act upon that group. However in this case its not as egregous as i've seen elsewhere so i'll leave it up to the PR author.

Reply:
Thanks both. I updated it to "managed_resource_group_name" and updated the document for the description.